doxygen/CMakeLists.txt

```diff
@@ -10,7 +10,14 @@ function(append_string_to_first_list_element APPEND_ARGS_INPUT APPEND_ARGS_APPEN
   set(${APPEND_OUT} ${APPEND_FINAL} PARENT_SCOPE)
 endfunction()
 
-set(DOXYGEN_OUTPUT_DIR ${CMAKE_BINARY_DIR}/doxygen)
+if (NOT DEFINED DOXYGEN_OUTPUT_DIR)
+  set(DOXYGEN_OUTPUT_DIR ${CMAKE_BINARY_DIR}/doxygen)
+else()
+  message("Doxygen output directory defined as: ${DOXYGEN_OUTPUT_DIR}")
+  set(DOXYGEN_OUTPUT_DIRECTORY ${DOXYGEN_OUTPUT_DIR})
+  set(DOXYGEN_HTML_OUTPUT ${DOXYGEN_OUTPUT_DIR})
+endif()
+
```

Comment:
Funnily enough, we use `DOXYGEN_OUTPUT_DIR` as our own variable, but we never used to set `DOXYGEN_OUTPUT_DIRECTORY` which is what CMake uses to define where everything should go. Was this intended?

(Logic behind this is that this now allows controlling where the files go - for us, I can specify the `/docs` folder so once pushed, the gh-pages branch should be happy with the docs folder)

Reply:
It probably should have been `DOXYGEN_OUTPUT_DIRECTORY` in the first place - I'm personally fine with doing a rename instead of anything strange like this

Reply:
Moved all to `DOXYGEN_OUTPUT_DIRECTORY` - just kept the step that reports if it is already defined (it will print it so we know where the output is)